packages/app/src/vis-packs/nexus/models.ts

```diff
@@ -23,39 +23,43 @@ export type NxAttribute =
   | 'SILX_style'
   | 'auxiliary_signals';
 
-export type AuxDatasets = {
-  signal: NumArrayDataset;
-  errors?: NumArrayDataset;
-}[];
+export interface DatasetInfo {
+  label: string;
+  unit: string | undefined;
+}
+
+interface DatasetDef<
+  T extends NumericType | ComplexType = NumericType | ComplexType
+> extends DatasetInfo {
+  dataset: Dataset<ArrayShape, T>;
+}
+
+type WithError<T extends DatasetDef> = T & {
+  errorDataset?: NumArrayDataset;
+};
+
+export type AuxDef = WithError<DatasetDef<NumericType>>;
```

Comment:
The chained generic are not very readable in my opinion but I don't see an easy alternative here

Reply:
Yeah I hesitated with `DatasetWithErrorDef extends `DatasetDef` but thought the mixin-style type was actually a better pattern in this case. `DatasetWithErrorDef` would have required duplicating the `T extends NumericType | ComplexType = NumericType | ComplexType` generic.